src/ngraph/runtime/cpu/builder/quantized_dot.cpp

```diff
@@ -114,70 +115,50 @@ namespace ngraph
-                                    deps,
-                                    ip_index,
-                                    arg0_buffer_index,
-                                    arg1_buffer_index,
-                                    arg2_buffer_index,
-                                    out0_buffer_index](CPURuntimeContext* ctx,
-                                                       CPUExecutionContext* ectx) mutable {
-                        if (ctx->first_iteration)
-                        {
-                            vector<float> dyn_scales;
-                            dyn_scales.assign(
-                                static_cast<float*>(ctx->buffer_data[arg2_buffer_index]),
-                                static_cast<float*>(ctx->buffer_data[arg2_buffer_index]) +
-                                    scales_size);
-                            ip_attr.set_output_scales(0, dyn_scales);
-                            mkldnn_emitter->build_inner_product_forward<false>(
-                                ctx->mkldnn_primitives,
-                                ip_desc,
-                                ip_attr,
-                                executor::global_cpu_engine,
-                                deps,
-                                ip_index);
-                        }
-                        cpu::mkldnn_utils::set_memory_ptr(
-                            ctx, deps[0], ctx->buffer_data[arg0_buffer_index]);
-                        cpu::mkldnn_utils::set_memory_ptr(
-                            ctx, deps[1], ctx->buffer_data[arg1_buffer_index]);
-                        cpu::mkldnn_utils::set_memory_ptr(
-                            ctx, deps[2], ctx->buffer_data[out0_buffer_index]);
-                        cpu::mkldnn_utils::mkldnn_invoke_primitive(ctx, ip_index);
-                    };
-                    functors.emplace_back(functor);
-                }
-                else
-                {
-                    throw ngraph_error("unsupported parameters for QuantizedDot via DEX");
-                }
+                auto scales_size = shape_size(args[2].get_shape());
+                std::function<decltype(
+                    runtime::cpu::kernel::dot_ref<uint8_t, uint8_t, uint8_t, int32_t>)>
+                    kernel;
+
+                kernel = runtime::cpu::kernel::dot_ref<uint8_t, uint8_t, uint8_t, int32_t>;
+
+                auto functor = [&,
+                                kernel,
+                                arg0_shape,
+                                arg1_shape,
+                                result_shape,
+                                arg0_buffer_index,
+                                arg1_buffer_index,
+                                arg2_buffer_index,
+                                out0_buffer_index,
+                                scales_size](CPURuntimeContext* ctx, CPUExecutionContext* ectx) {
+
+                    vector<float> dyn_scales;
```

Comment:
use a straight float instead of vector since scales_size is always going to be 1.